monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/python
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for check_collectd_mlab check."""
+
+import os
+import socket
+import time
+import unittest2 as unittest
+
+# Third-party modules.
+import mock
+
+# Module under test.
+import check_collectd_mlab
+
+# R0904: Too many public methods. Hard to avoid for unit tests.
+# pylint: disable=R0904
+
+
+class MLabNagiosSocketTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(check_collectd_mlab.__file__), 'testdata')
+
+  def testunit_sock_sendcmd_RETURNS_successfully(self):
+    mock_sock = mock.Mock(spec_set=socket.socket)
+    mock_sock.recv.side_effect = list('1 default reply\n')
+
+    returned_value = check_collectd_mlab.sock_sendcmd(
+        mock_sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 1)
+    mock_sock.send.assert_called_with('GETVAL "whatever"\n')
+
+  def testunit_sock_sendcmd_WHEN_receive_bad_reply_RETURNS_zero(self):
+    mock_sock = mock.Mock(spec_set=socket.socket)
+    mock_sock.recv.side_effect = list('not-a-number junk\n')
+
+    returned_value = check_collectd_mlab.sock_sendcmd(
+        mock_sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 0)
+    mock_sock.send.assert_called_with('GETVAL "whatever"\n')
+
+  def testunit_sock_connect_WHEN_no_socket_RAISES_Error(self):
```

Comment:
It looks like the original form was "_RAISES_CriticalError" and when I generalized the exception names to not reference "Critical" these became "_RAISES_Error". Originally, the name / capitalization referenced an actual type so that seemed to make sense, rather than un-capitalizing it. But, since Error doesn't refer to an actual type now, I changed the capitalization to lowercase.  4d04ea89f6aaf748520be3f25ecc45dbd54e8162